arch/arm/core/aarch32/cortex_a_r/exc.S

```diff
@@ -61,7 +61,21 @@ SECTION_SUBSEC_FUNC(TEXT, __exc, z_arm_undef_instruction)
 	 * Store r0-r3, r12, lr, lr_und and spsr_und into the stack to
 	 * construct an exception stack frame.
 	 */
-	srsdb sp, #MODE_UND!
+	srsdb sp!, #MODE_UND
+	stmfd sp, {r0-r3, r12, lr}^
+	sub sp, #24
```

Comment:
Not sure will it be better to pack these similar codes into a macro or something in the future?  Anyway, LGTM.
My only concern maybe is (not sure because it's only related to exceptions, abort, etc.), since it will always cost double the space of the stack, will it cause some kind of performance problems?

Reply:
It looks like the undef exception could potentially be problematic in the future since it could be used for emulating instructions; but, IMO, that that's something for a future PR that actually implements such features.